sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -201,13 +201,14 @@ case class AlterTableAddColumnsCommand(
 
     // make sure any partition columns are at the end of the fields
     val reorderedSchema = catalogTable.dataSchema ++ columns ++ catalogTable.partitionSchema
+    val newSchema = catalogTable.schema.copy(fields = reorderedSchema.toArray)
 
     SchemaUtils.checkColumnNameDuplication(
       reorderedSchema.map(_.name), "in the table definition of " + table.identifier,
       conf.caseSensitiveAnalysis)
+    DDLUtils.checkDataSchemaFieldNames(catalogTable.copy(schema = newSchema))
```

Comment:
`newSchema ` also contains `partition schema`. How about partition schema? Do we have the same limits on it?

Reply:
It's okay. Inside `checkDataSchemaFieldNames`, we only uses `table.dataSchema` like the following.
```
ParquetSchemaConverter.checkFieldNames(table.dataSchema)
```

For the partition columns, we have been allowing the special characters.

Reply:
Could you add test cases and ensure the partitioning columns with special characters work?

Reply:
DDLSuite and HiveDDLSuite have them here.
  - https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala#L2045-L2070
  - https://github.com/apache/spark/blob/master/sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala#L1736

Reply:
I think this PR passes the above two test cases, too.